src/App.tsx

```diff
@@ -1,27 +1,101 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('0');
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+  const handleAllTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let hasError = false;
+
+    if (title.trim() === '') {
+      setTitleError('Please enter a title');
+      hasError = true;
+    } else {
+      setTitleError('');
+    }
+
+    if (selectedUser === '0') {
+      setUserError('Please choose a user');
+      hasError = true;
+    } else {
+      setUserError('');
+    }
+
+    if (!hasError) {
+      const newTodo = {
+        id: todos.length ? Math.max(...todos.map(todo => todo.id)) + 1 : 1,
+        title: title.trim(),
+        userId: Number(selectedUser),
```

Comment:
Ensure that `selectedUser` is always a valid user ID. Currently, it defaults to '0', which is not a valid user ID. Consider initializing `selectedUser` with a valid user ID or handling the case where no valid user is selected more explicitly.